crates/re_viewport/src/context_menu.rs

```diff
@@ -101,6 +132,59 @@ pub fn context_menu_ui_for_item(
     });
 }
 
+// ================================================================================================
+// Selection summary
+// ================================================================================================
+
+// TODO(ab): this summary is somewhat ad hoc to the context menu needs. Could it be generalised and
+// moved to the Selection itself?
+#[derive(Debug, Clone)]
+pub enum SelectionSummary {
+    SingleContainerItem(ContainerId),
+    ContentsItems(Vec<Contents>),
+    Heterogeneous,
+    Empty,
+}
+
+fn summarize_selection(selection: &Selection) -> SelectionSummary {
+    if selection.is_empty() {
+        return SelectionSummary::Empty;
+    }
+
+    if selection.len() == 1 {
+        if let Some(Item::Container(container_id)) = selection.first_item() {
+            return SelectionSummary::SingleContainerItem(*container_id);
+        }
+    }
+
+    // test if the selection contains only contents
+    let mut only_space_view_or_container_only = true;
+    for (item, _) in selection.iter() {
+        match item {
+            Item::Container(_) | Item::SpaceView(_) => {}
+            _ => only_space_view_or_container_only = false,
+        }
+    }
+
+    if only_space_view_or_container_only {
+        let mut contents = vec![];
+        for (item, _) in selection.iter() {
+            match item {
+                Item::Container(container_id) => {
+                    contents.push(Contents::Container(*container_id));
+                }
+                Item::SpaceView(space_view_id) => {
+                    contents.push(Contents::SpaceView(*space_view_id));
+                }
+                _ => unreachable!(),
+            }
+        }
```

Comment:
suggestion in pseudo-code:
```rust
let contents = selection.filter_map(match { ... }).collect();
```
which gets rid of the `unreachable` that should not be here in release builds